addons/hr_expense/wizard/account_payment_register.py

```diff
@@ -24,6 +24,10 @@ def _create_payments(self):
         payments = super()._create_payments()
 
         expense_sheets = self.env['hr.expense.sheet'].search([('account_move_id', 'in', self.line_ids.move_id.ids)])
+        for l in payments.move_id.line_ids:
+            if expense_sheets.expense_line_ids:
+                l.expense_id = expense_sheets.expense_line_ids[0]
```

Comment:
The purpose of this method is to create the payments, not to assign some values randomly. I don't understand what you are trying to do nor how is it fixing an issue. Plz add a test showcasing the issue